### PR TITLE
Improvements to Faker.Address.Es Module

### PR DESCRIPTION
**Pull request by emig**
Added: region function
Added:
  * complete the list of spanish regions
  * document source

Fix:
  * use spanish names for street and city names
  * use correct prefixes and siffixes for stret names

I've added:

- [x] USAGE.md docs if applicable
- [x] CHANGELOG.md

### CODE REVIEW

File: lib/faker/address/es.ex [hunk 1]
```diff
@@ -429,37 +429,44 @@ defmodule Faker.Address.Es do
...
-      iex> Faker.Address.Es.state()
-      "Comunidad Valenciana"
-      iex> Faker.Address.Es.state()
-      "Galicia"
-      iex> Faker.Address.Es.state()
-      "Cataluña"
+      iex> Faker.Address.Es.region()
+      "Extremadura"
+      iex> Faker.Address.Es.region()
+      "Aragón"
```
Comment (vbrazo) [inline lib/faker/address/es.ex]: As standard pattern, we need 4 examples in the doctests.

Comment (vbrazo) [inline lib/faker/address/es.ex]: As a pattern we use 4 examples in the doctests.

File: lib/faker/address/es.ex [hunk 2]
```diff
@@ -429,37 +429,44 @@ defmodule Faker.Address.Es do
...
   """
-  @spec state() :: String.t()
-  sampler(:state, [
+
+  @spec region() :: String.t()
+  sampler(:region, [
```
Comment (anthonator) [inline lib/faker/address/es.ex]: Are we cool with deviating from the other address modules? The others don’t have a region function.

File: lib/faker/address/es.ex (hunk 1 above)
Comment (emig) [reply lib/faker/address/es.ex]: Thanks, I'll add them

File: lib/faker/address/es.ex (hunk 2 above)
Comment (emig) [reply lib/faker/address/es.ex]: I've followed the italian version when implementing this https://github.com/elixirs/faker/blob/4ec3a668ec350e8846c786a8b3fcc8e0b9ce4a89/lib/faker/address/it.ex#L962

Comment (anthonator) [reply lib/faker/address/es.ex]: I didn’t see that in the Italian translations. Thanks for pointing that out.

File: lib/faker/address/es.ex [hunk 3]
```diff
@@ -429,37 +429,48 @@ defmodule Faker.Address.Es do
   end
 
   @doc """
-  Return state.
+  Return state. But Spain doesn't has states so this calls Faker.Address.Es.region() instead
```
Comment (vbrazo) [inline lib/faker/address/es.ex]:
```suggestion
  Return state. But Spain doesn't have states so this calls Faker.Address.Es.region() instead.
```

**Pull request merged**